crates/cxx-qt-lib/src/core/qtlogging.rs

```diff
@@ -0,0 +1,138 @@
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Joshua Goins <joshua.goins@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+use cxx::{type_id, ExternType};
+use std::ffi::c_char;
+use std::ffi::CStr;
+
+#[cxx::bridge]
+mod ffi {
+    #[repr(i32)]
+    enum QtMsgType {
+        QtDebugMsg = 0,
+        QtInfoMsg = 4,
+        QtWarningMsg = 1,
+        QtFatalMsg = 3,
+        QtCriticalMsg = 2,
+        QtSystemMsg = 2
+    }
+
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/common.h");
+
+        include!("cxx-qt-lib/qstring.h");
+        type QString = crate::QString;
+
+        include!("cxx-qt-lib/qtlogging.h");
+        type QMessageLogContext = crate::QMessageLogContext;
+        type QtMsgType;
+
+        fn qt_message_output(msgType: QtMsgType, context: &QMessageLogContext, string: &QString);
+
+        #[cxx_name = "qmessagelogcontext_line"]
+        fn line(context: &QMessageLogContext) -> i32;
```

Comment:
you should be able to avoid the C++ proxy for types that CXX can express here and just do

```rust
fn line(self: &QMessageContext) -> i32;
```

Then you can remove the ffi::line() call below.

Reply:
ah i see there are no actual member functions in QMessageLogContext :see_no_evil: so you can't :sweat_smile: